pkg/tsdb/loki/loki.go

```diff
@@ -122,22 +123,25 @@ func (s *Service) CallResource(ctx context.Context, req *backend.CallResourceReq
 func getHeadersForCallResource(headers map[string][]string) map[string]string {
 	data := make(map[string]string)
 
-	if auth := arrayHeaderFirstValue(headers["Authorization"]); auth != "" {
-		data["Authorization"] = auth
-	}
-
-	if cookie := arrayHeaderFirstValue(headers["Cookie"]); cookie != "" {
-		data["Cookie"] = cookie
-	}
-
-	if idToken := arrayHeaderFirstValue(headers["X-ID-Token"]); idToken != "" {
-		data["X-ID-Token"] = idToken
-	}
+	for k, values := range headers {
+		k = textproto.CanonicalMIMEHeaderKey(k)
+		firstValue := arrayHeaderFirstValue(values)
 
-	if encType := arrayHeaderFirstValue(headers["Accept-Encoding"]); encType != "" {
-		data["Accept-Encoding"] = encType
+		if firstValue == "" {
+			continue
+		}
+		switch k {
+		case "Authorization":
+			data["Authorization"] = firstValue
+		case "X-Id-Token":
+			data["X-ID-Token"] = firstValue
+		case "Cookie":
+			data["Cookie"] = firstValue
+		}
+    case "Accept-Encoding":
+			data["Accept-Encoding"] = firstValue
+		}
```

Comment:
I think this needs to move up one line, right?

Reply:
Yes, sorry I missed that when merging and then got distracted.
I also fixed the test that was failing after the merge and added the `Accept-Encoding` header to the test case